server/resp/src/main/java/org/infinispan/server/resp/commands/json/JSONSET.java

```diff
@@ -0,0 +1,141 @@
+
+   public static final String XX = "XX";
+   public static final String NX = "NX";
+   // Configure JsonPath to use Jackson. Path with missing final leaf will return null
+   // path with more the 1 level missing will rise exception
+   static Configuration config = Configuration.builder().options(Option.DEFAULT_PATH_LEAF_TO_NULL)
+         .jsonProvider(new com.jayway.jsonpath.spi.json.JacksonJsonNodeJsonProvider())
+         .mappingProvider(new com.jayway.jsonpath.spi.mapper.JacksonMappingProvider())
+         .build();
+
+   public static void jsonSetBiConsumer(CharSequence value, ByteBufPool alloc) {
+      if (value == null) {
+         // Use null consumer
+         Resp3Response.nulls(alloc);
+         return;
+      }
+      if (value.length() == 2 && value.charAt(0) == 'O' && value.charAt(1) == 'K') {
+         // Use ok consumer
+         Resp3Response.ok(alloc);
+         return;
+      }
+      // All the rest are errors
+      RespErrorUtil.customError(value.toString(), alloc);
+   }
+
+   public JSONSET() {
+      super("JSON.SET", -3, 1, 1, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler, ChannelHandlerContext ctx,
+         List<byte[]> arguments) {
+      byte[] key = arguments.get(0);
+      byte[] path = arguments.get(1);
+      byte[] value = arguments.get(2);
+      if (arguments.size() > 4) {
+         RespErrorUtil.syntaxError(handler.allocator());
+         return handler.myStage();
+      }
+      boolean nx, xx; // must be effectively final
+      if (arguments.size() == 4) {
+         String arg = (new String(arguments.get(3)).toUpperCase());
+         switch (arg) {
+            case NX:
+               nx = true;
+               xx = false;
+               break;
+            case XX:
+               xx = true;
+               nx = false;
+               break;
+            default:
+               RespErrorUtil.syntaxError(handler.allocator());
+               return handler.myStage();
+         }
+      } else {
+         nx = false;
+         xx = false;
+      }
+      ObjectMapper mapper = new ObjectMapper();
+         JsonNode newNode;
+         try {
+            newNode = mapper.readTree(value);
+         } catch (IOException e) {
+            RespErrorUtil.customError(e.getMessage(), handler.allocator());
+            return handler.myStage();
+         }
+         FunctionalMap.ReadWriteMap<byte[], Object> cache = ReadWriteMapImpl
+               .create(FunctionalMapImpl.create(handler.typedCache(null)));
+         CompletionStage<CharSequence> cs = cache.eval(key, view -> {
+            var doc = (byte[]) view.find().orElse(null);
+            if (doc == null) {
+               if (xx) {
+                  return null;
+               }
+               if (!isRoot(path)) {
+                  return "new objects must be created at root";
+               }
+               view.set(value);
+               return "OK";
```

Comment:
Can we just return a Boolean instead? It is confusing seeing "OK" and RespContants.OK as return values. The boolean would just be was it written or not and the exception would just be propagated to the stage and handled as needed.

Reply:
I cleand up a bit, I tried to avoid exceptions on user input because they bubble up to the log and we already have this: [ISPN-16783](https://issues.redhat.com/browse/ISPN-16783) [RESP] Error stacktrace on WRONGTYPE

Reply:
mmm maybe we can add some code here and handle an ad hoc the exception: https://github.com/infinispan/infinispan/blob/1c83c4ef8f1506c1136d273264b5eb0a5a5898fe/server/resp/src/main/java/org/infinispan/server/resp/serialization/ResponseWriter.java#L159